src/main/java/com/zenuml/confluence/macro/SequenceMacro.java

```diff
@@ -34,27 +34,27 @@ public SequenceMacro(
         this.settingsManager = settingsManager;
     }
 
-    public String createMD5(String plaintext) {
+    private String createMD5(String plaintext) {
         try {
             MessageDigest md = MessageDigest.getInstance("MD5");
             md.update(plaintext.getBytes());
             byte[] digest = md.digest();
             String myHash = DatatypeConverter.printHexBinary(digest).toLowerCase();
             return  myHash;
         } catch (NoSuchAlgorithmException e) {
-            return null;
+            throw new RuntimeException(e);
         }
     }
 
-    public String getPDFExportImgTag(String dsl, ConversionContext conversionContext) {
+    private String getPDFExportImgTag(String dsl, ConversionContext conversionContext) {
         String baseUrl = settingsManager.getGlobalSettings().getBaseUrl();
         String hash = createMD5(dsl);
         if(hash == null){
```

Comment:
Will hash ever be null? It is not necessary to do nullable check again. The createMD5 method (as a private method) should ensure it either returns a valid hash or throw an exception.

Reply:
String Class has hashCode() method. should we use it or create our own version?